tests/upgrades/test_syncplan.py

```diff
@@ -14,100 +14,88 @@
+    )[0]
+    assert product.sync_plan.id == sync_plan.id
+    assert sync_plan.name == f'{pre_test_name}_syncplan'
+    assert sync_plan.interval == 'hourly'
+    for sync_interval in SYNC_INTERVAL:
+        if sync_interval == "custom":
+            sync_plan.interval = SYNC_INTERVAL['custom']
+            sync_plan.cron_expression = gen_choice(valid_cron_expressions())
+            sync_plan.update(['interval', 'cron_expression'])
+        else:
+            sync_plan.interval = SYNC_INTERVAL[sync_interval]
+            sync_plan.update(['interval'])
         sync_plan = sync_plan.read()
-        scenario_dict = {
-            self.__class__.__name__: {
-                'sync_plan_name': sync_plan_name,
-                'interval': sync_plan.interval,
-                'sync_date': sync_plan.sync_date,
-                'product_id': product.id,
-                'sync_plan_id': sync_plan.id,
-                'org_id': org.id,
-            }
-        }
-        create_dict(scenario_dict)
-
-    @post_upgrade(depend_on=test_pre_sync_plan_migration)
-    def test_post_sync_plan_migration(self):
-        """Post-upgrade scenario that tests existing sync plans are working as
-        expected after satellite upgrade with migrating from pulp to katello
-
-        :id: badaeec2-d42f-41d5-bd85-4b23d6d5a724
-
-        :steps:
-            1. Verify sync plan exists and works as earlier
-
-        :expectedresults: Post Upgrade, Sync plans exists and works as earlier.
-
-        """
-        beforeupgrade_data = get_entity_data(self.__class__.__name__)
-        org = entities.Organization(id=beforeupgrade_data.get('org_id'))
-        product = entities.Product(id=beforeupgrade_data.get("product_id")).read()
-        sync_plan = entities.SyncPlan(
-            id=beforeupgrade_data.get("sync_plan_id"), organization=org
-        ).read()
-        self.assertEqual(product.sync_plan.id, sync_plan.id)
-        self.assertEqual(sync_plan.name, beforeupgrade_data.get("sync_plan_name"))
-        self.assertEqual(sync_plan.interval, beforeupgrade_data.get("interval"))
-        self.assertEqual(sync_plan.sync_date, beforeupgrade_data.get("sync_date"))
-        # checking sync plan update on upgraded satellite
-        sync_plan.interval = SYNC_INTERVAL['custom']
-        sync_plan.cron_expression = gen_choice(valid_cron_expressions())
-        self.assertEqual(
-            sync_plan.update(['interval', 'cron_expression']).interval, SYNC_INTERVAL['custom']
-        )
-        # checking sync plan delete on upgraded satellite
-        sync_plan.delete()
-        product = product.read()
-        self.assertIsNone(product.sync_plan)
-        with self.assertRaises(HTTPError):
-            sync_plan.read()
+        assert sync_plan.interval == SYNC_INTERVAL[sync_interval]
+    sync_plan.delete()
+    product = product.read()
+    assert product.sync_plan is None
+    with pytest.raises(HTTPError):
+        sync_plan.read()
+    product.delete()
```

Comment:
If these deletes aren't really part of the test coverage, consider moving them into the `teardown` phase by wrapping their call with `request.addfinalizer(product.delete)` and move the call up in the test function to right after the `product` is created.

Reply:
Fixed